File: examples/test_transport.py
#!/usr/bin/env python3
"""
Test script for Janus ZeroMQ Transport Plugin

This script demonstrates how to communicate with Janus via ZeroMQ transport.
"""

import zmq
import json
import sys

def test_transport():
    """Test the ZeroMQ transport plugin"""
    print("Testing Janus ZeroMQ Transport...")
    
    # Create ZeroMQ context and socket
    context = zmq.Context()
    socket = context.socket(zmq.REQ)
    
    try:
        # Connect to Janus
        socket.connect("tcp://127.0.0.1:5545")
        print("✓ Connected to tcp://127.0.0.1:5545")
        
        # Set timeout
        socket.setsockopt(zmq.RCVTIMEO, 5000)  # 5 seconds
        
        # Test 1: Get server info
        print("\nTest 1: Get server info")
        request = {
            "janus": "info",
            "transaction": "test-info"
        }
        print(f"Sending: {json.dumps(request, indent=2)}")
        socket.send_string(json.dumps(request))
        
        response = socket.recv_string()
        response_data = json.loads(response)
        print(f"Response: {json.dumps(response_data, indent=2)}")
        
        if response_data.get("janus") == "server_info":
            print("✓ Server info received successfully")
        else:
            print("✗ Unexpected response")
            return False
        
        # Test 2: Create session
        print("\nTest 2: Create session")
        request = {
            "janus": "create",
            "transaction": "test-create"
        }
        print(f"Sending: {json.dumps(request, indent=2)}")
        socket.send_string(json.dumps(request))
        
        response = socket.recv_string()
        response_data = json.loads(response)
        print(f"Response: {json.dumps(response_data, indent=2)}")
        
        if response_data.get("janus") == "success":
            session_id = response_data.get("data", {}).get("id")
            print(f"✓ Session created: {session_id}")
        else:
            print("✗ Failed to create session")
            return False
        
        print("\n✓ All tests passed!")
        return True
        
    except zmq.error.Again:
        print("✗ Timeout waiting for response. Is Janus running with ZeroMQ transport enabled?")
        return False
    except Exception as e:
        print(f"✗ Error: {e}")
        return False
    finally:
        socket.close()
        context.term()

if __name__ == "__main__":
    success = test_transport()
    sys.exit(0 if success else 1)

File: examples/test_transport.js
#!/usr/bin/env node
/**
 * Test script for Janus ZeroMQ Transport Plugin (Node.js)
 * 
 * This script demonstrates how to communicate with Janus via ZeroMQ transport.
 * 
 * Installation: npm install zeromq
 */

const zmq = require('zeromq');

async function testTransport() {
    console.log('Testing Janus ZeroMQ Transport...');
    
    // Create REQ socket
    const socket = new zmq.Request();
    
    try {
        // Connect to Janus
        socket.connect('tcp://127.0.0.1:5545');
        console.log('✓ Connected to tcp://127.0.0.1:5545');
        
        // Set timeout
        socket.receiveTimeout = 5000;
        
        // Test 1: Get server info
        console.log('\nTest 1: Get server info');
        const infoRequest = {
            janus: 'info',
            transaction: 'test-info'
        };
        console.log('Sending:', JSON.stringify(infoRequest, null, 2));
        await socket.send(JSON.stringify(infoRequest));
        
        const [infoResponse] = await socket.receive();
        const infoData = JSON.parse(infoResponse.toString());
        console.log('Response:', JSON.stringify(infoData, null, 2));
        
        if (infoData.janus === 'server_info') {
            console.log('✓ Server info received successfully');
        } else {
            console.log('✗ Unexpected response');
            return false;
        }
        
        // Test 2: Create session
        console.log('\nTest 2: Create session');
        const createRequest = {
            janus: 'create',
            transaction: 'test-create'
        };
        console.log('Sending:', JSON.stringify(createRequest, null, 2));
        await socket.send(JSON.stringify(createRequest));
        
        const [createResponse] = await socket.receive();
        const createData = JSON.parse(createResponse.toString());
        console.log('Response:', JSON.stringify(createData, null, 2));
        
        if (createData.janus === 'success') {
            const sessionId = createData.data.id;
            console.log(`✓ Session created: ${sessionId}`);
        } else {
            console.log('✗ Failed to create session');
            return false;
        }
        
        console.log('\n✓ All tests passed!');
        return true;
        
    } catch (error) {
        console.error('✗ Error:', error.message);
        return false;
    } finally {
        socket.close();
    }
}

// Run test
testTransport().then(success => {
    process.exit(success ? 0 : 1);
}).catch(error => {
    console.error('Fatal error:', error);
    process.exit(1);
});
